### DISCUSSION TITLE
Tiled VAE Encode for inpaint?

### DISCUSSION THREAD

**Discussion by okolenmi** [Comfy-Org/ComfyUI#1393] [Ideas]
Sadly, I can't use inpaint on images 1.5MPixels+. It looks like I need at least 6GB VRAM to pass VAE Encode (for inpainting) step on 1920*1080 image. This step on my CPU only is about 40 seconds, but Sampler processing is about 3 hours. It was used for outpainting.

--- 
![изображение](https://github.com/comfyanonymous/ComfyUI/assets/55319814/7681209d-0508-4f4e-bcfc-7f588fc59048)
In this case I have a fail - black border without a result (it happens sometimes...)
---

So, I need to pass Encoding step somehow to process such images on my GPU for 10x+ speed up. Does the developer plan to make a tiled version of this encoder or not? Or is there a way to switch this step to CPU processing?

**Comment by comfyanonymous** [comment]
You can try: --fp16-vae if the VAE isn't one of the ones that has black image issues when running in fp16 it will reduce vram usage.
